pyquil/api/_compiler.py

```diff
@@ -160,6 +166,11 @@ def native_quil_to_executable(self, nq_program: Program) -> BinaryExecutableResp
                           "Program that hasn't been compiled via `quil_to_native_quil`. This is "
                           "ok if you've hand-compiled your program to our native gateset, "
                           "but be careful!")
+        if self.name is not None:
+            targeted_lattice = self.client.call('get_lattice_name')
+            if self.client.call('get_lattice_name') != self.name:
```

Comment:
You might also want to guard this against the compiler server returning `None`. That's not a consideration for external users in their normal mode of operation, but internal users do also use pyQuil against local compiler-server instances. Otherwise (and pending action on the internal PR review), this looks fine to me.